despatch_advice_import/wizard/despatch_advice_import.py

```diff
@@ -136,16 +136,22 @@ def process_document(self):
     def process_data(self, parsed_order_document):
         bdio = self.env["business.document.import"]
         po_name = parsed_order_document.get("ref")
-        order = self.env["purchase.order"].search([("name", "=", po_name)])
-
-        if not order:
-            bdio.user_error_wrap(_("No purchase order found for name %s.") % po_name)
 
         lines = parsed_order_document.get("lines")
         lines_by_id = {int(line["line_id"]): line for line in lines}
 
         for line_id, line_info in lines_by_id.iteritems():
-            line = order.order_line.search([("id", "=", line_id)])
+            line = self.env["purchase.order.line"].search([("id", "=", line_id)])
+
+            # Check PO exist
+            order = self.env["purchase.order"].search([("name", "=", line_info["ref"])])
```

Comment:
The order is available on the line `order = line.order_id`. Yous should only check:
 * if the order ref is provided, the line.order_id.ref must == order_ref
 * if no order ref provided: line.order_id.name == po_name